clients/src/main/java/org/apache/kafka/common/requests/ElectPreferredLeadersRequest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.requests;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.types.ArrayOf;
+import org.apache.kafka.common.protocol.types.Field;
+import org.apache.kafka.common.protocol.types.Schema;
+import org.apache.kafka.common.protocol.types.Struct;
+import org.apache.kafka.common.utils.CollectionUtils;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.kafka.common.protocol.CommonFields.TOPIC_NAME;
+import static org.apache.kafka.common.protocol.types.Type.INT32;
+
+public class ElectPreferredLeadersRequest extends AbstractRequest {
+
+    private static final String TIMEOUT_KEY_NAME = "timeout";
+    private static final String TOPIC_PARTITIONS_KEY_NAME = "topic_partitions";
+    private static final String PARTITIONS_KEY_NAME = "partitions";
+
+    public static final Schema ELECT_PREFERRED_LEADERS_REQUEST_V0 = new Schema(
+            new Field(TOPIC_PARTITIONS_KEY_NAME, ArrayOf.nullable(
+                    new Schema(
+                            TOPIC_NAME,
+                            new Field(PARTITIONS_KEY_NAME,
```

Comment:
Could we reuse PARTITION_ID in CommonFields?

Reply:
`PARTITION_ID` is a single partition, but `PARTITIONS_KEY_NAME` is an array of partitions.